File: front-end/src/app/dto/response/discipline-info-response.ts
import {ReviewResponse} from "./review-response";
import {DisciplineCategory} from "../../vo/discipline-category";
import {DisciplineType} from "../../vo/discipline-type";

export class DisciplineInfoResponse {
    id: string;
    name: string;
    credits: number;
    category: DisciplineCategory;
    type: DisciplineType;
    reviews: ReviewResponse[];
}

File: src/main/java/bg/fmi/rateuni/dto/response/DisciplineInfoResponse.java
package bg.fmi.rateuni.dto.response;

import bg.fmi.rateuni.vo.DisciplineCategory;
import bg.fmi.rateuni.vo.DisciplineType;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
public class DisciplineInfoResponse {
    private UUID id;
    private String name;
    private double credits;
    private String description;
    private DisciplineCategory category;
    private DisciplineType type;
    private List<ReviewResponse> reviews;
}
